packages/test/test-end-to-end-tests/src/test/gc/gcVersionUpdate.spec.ts

```diff
@@ -98,14 +100,27 @@ describeCompat("GC version update", "NoCompat", (getTestObjectProvider, apis) =>
 		return summaryResult.summaryVersion;
 	}
 
+	/**
+	 * Generates a summary and validates that the GC state is reset (empty) in the base GC details. This will ensure
+	 * that the GC data is regenerated by all the nodes.
+	 */
+	async function summarizeAndValidateGCStateReset(summarizer: ISummarizer) {
+		const containerRuntime = (summarizer as any).runtime as IContainerRuntimeWithPrivates;
+		const spy = sandbox.spy(containerRuntime.garbageCollector, "getBaseGCDetails");
```

Comment:
What does the sandbox object do exactly? Just lets you spy on stuff?

Reply:
I think so. That among other things. I just copied the pattern from other tests.